platform-sdk/swirlds-logging-log4j-appender/src/test/java/com/swirlds/logging/log4j/appender/SwirldsLogAppenderTest.java

```diff
@@ -0,0 +1,191 @@
+                Arguments.of(Level.DEBUG, 4),
+                Arguments.of(Level.INFO, 3),
+                Arguments.of(Level.WARN, 2),
+                Arguments.of(Level.ERROR, 1),
+                Arguments.of(Level.OFF, 0));
+    }
+
+    @ParameterizedTest
+    @MethodSource("levelsAndSizes")
+    void loggingLevels(final Level level, final int expectedSize) throws Exception {
+        // given
+        final Path filePath = Path.of("%s_test.log".formatted(level));
+        try {
+            final LoggingSystem loggingSystem = setup(createConfig(level, filePath));
+            final Logger testLogger = LogManager.getLogger(LOGGER_NAME);
+
+            // when
+            testLogger.error("This is an error message");
+            testLogger.warn("This is a warn message");
+            testLogger.info("This is an info message");
+            testLogger.debug("This is a debug message");
+            testLogger.trace("This is a trace message");
+
+            // then
+            loggingSystem.stopAndFinalize();
+            final List<String> logLines = Files.lines(filePath).toList();
+
+            assertThat(logLines).hasSize(expectedSize);
+        } finally {
+            Files.delete(filePath);
+        }
+    }
+
+    @Test
+    void loggingSimpleMarker() throws Exception {
+        // given
+        final Path filePath = Path.of("marker_test.log");
+        try {
+            final LoggingSystem loggingSystem = setup(createConfig(Level.DEBUG, filePath));
+            final Logger testLogger = LogManager.getLogger(LOGGER_NAME);
+
+            // when
+            testLogger.warn(LogMarker.CONFIG.getMarker(), "This is a warn message");
+
+            // then
+            loggingSystem.stopAndFinalize();
+            final List<String> logLines = Files.lines(filePath).toList();
+
+            assertThat(logLines).hasSize(1);
+            assertThat(logLines.getFirst()).contains(LogMarker.CONFIG.name());
+        } finally {
+            Files.delete(filePath);
+        }
+    }
+
+    @Test
+    void loggingChainedMarkers() throws Exception {
+        // given
+        final Path filePath = Path.of("marker_test.log");
+        try {
+            final LoggingSystem loggingSystem = setup(createConfig(Level.DEBUG, filePath));
+            final Logger testLogger = LogManager.getLogger(LOGGER_NAME);
+
+            // when
+            testLogger.warn(TestMarkers.CHILD_MARKER, "This is a warn message");
+
+            // then
+            loggingSystem.stopAndFinalize();
+            final List<String> logLines = Files.lines(filePath).toList();
+
+            assertThat(logLines).hasSize(1);
+            assertThat(logLines.getFirst()).contains(TestMarkers.CHILD, TestMarkers.PARENT, TestMarkers.GRANT);
+        } finally {
+            Files.delete(filePath);
+        }
+    }
+
+    @Test
+    void loggingContext() throws Exception {
+        // given
+        final Path filePath = Path.of("context_test.log");
```

Comment:
instead of having a try / catch in every test that makes it hard to read you can use https://www.baeldung.com/junit-5-temporary-directory